backend/setup.go

```diff
@@ -66,3 +87,52 @@ func SetupPluginEnvironment(pluginID string) {
 		}()
 	}
 }
+
+// SetupTracer sets up the global OTEL trace provider and tracer.
+func SetupTracer(pluginID string, tracingOpts tracing.Opts) error {
```

Comment:
Is there a possibility that `tracingOpts` is nil and `tracingCfg.IsEnabled()` is true?

Reply:
tracingOpts is a struct value so it cannot be nil. It could be zero-value, but that's fine and it won´t cause any issues :smile: